scrapyd_k8s/log_handler.py

```diff
@@ -0,0 +1,54 @@
+import os
+from kubernetes import client, watch
+import subprocess
+
+def get_last_n_lines(file_path, num_lines_to_check):
+    lines = []
+    try:
+        with open(file_path, 'r') as file:
+            lines = file.readlines()
+        return lines[-num_lines_to_check:]
+    except FileNotFoundError:
+        return []
+
+def concatenate_and_delete_files(main_file_path, temp_file_path):
+    try:
+        subprocess.run(f"cat {temp_file_path} >> {main_file_path}", shell=True, check=True)
+        os.remove(temp_file_path)
+    except Exception as e:
+        print(f"CONCATENATION ERROR: {e}")
+
+def make_log_filename_for_job(job_name):
+    return os.path.join('/tmp', f"{job_name}_logs.txt")
```

Comment:
please use `mkstemp` and friends, or rather the Python alternative, to avoid naming race conditions (even though we're probably running in a container). Or perhaps a temporary directory to keep `/tmp` a bit clean. Also, look into standard environment variables to set the temporary directory (like `TMPDIR`, though ideally this would be handled by the Python tempfile-related standard library, not sure).